inc/class-shortcode-ui.php

```diff
@@ -68,6 +68,31 @@ private function setup_actions() {
 		add_action( 'admin_enqueue_scripts', array( $this, 'action_admin_enqueue_scripts' ) );
 		add_action( 'wp_enqueue_editor',     array( $this, 'action_wp_enqueue_editor' ) );
 		add_action( 'wp_ajax_do_shortcode',  array( $this, 'handle_ajax_do_shortcode' ) );
+		add_filter( 'wp_editor_settings',    array( $this, 'filter_wp_editor_settings' ), 10, 2 );
+	}
+
+	/**
+	 * When a WP_Editor is initialized on a page, call the 'register_shortcode_ui' action.
+	 *
+	 * This action can be used to register styles and shortcode UI for any
+	 * shortcake-powered shortcodes, only on views which actually include a WP
+	 * Editor.
+	 */
+	public function filter_wp_editor_settings( $settings, $editor_id ) {
+
+		if ( ! did_action( 'register_shortcode_ui' ) ) {
+
+			/**
+			 * Register shortcode UI for shortcodes.
+			 *
+			 * Can be used to register shortcode UI only when an editor is being enqueued.
+			 *
+			 * @param array $settings Settings array for the ective WP_Editor.
+			 */
+			do_action( 'register_shortcode_ui', $settings, $editor_id );
```

Comment:
Is there a reason you might want to register different shortcode UI for different editors?

We could make the action more specific to accommodate this (`register_shortcode_ui_for_editor`), and make it clear the action could be fired multiple times on one page.